Importando a biblioteca Pandas

In [148]:
import pandas as pd

Importando o DataFrame

In [149]:
df = pd.read_csv('/content/drive/MyDrive/Trabalho 3 ENG1111/class_analise_credito.csv')

2) Verificando as informações do DataFrame

In [150]:
print(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1500 entries, 0 to 1499
Data columns (total 12 columns):
 #   Column   Non-Null Count  Dtype
---  ------   --------------  -----
 0   ESTC     1500 non-null   int64
 1   NDEP     1500 non-null   int64
 2   RENDA    1500 non-null   int64
 3   TIPOR    1500 non-null   int64
 4   VBEM     1500 non-null   int64
 5   NPARC    1500 non-null   int64
 6   VPARC    1500 non-null   int64
 7   TEL      1500 non-null   int64
 8   IDADE    1500 non-null   int64
 9   RESMS    1500 non-null   int64
 10  ENTRADA  1500 non-null   int64
 11  target   1500 non-null   int64
dtypes: int64(12)
memory usage: 140.8 KB
None


3) Usei esse comando para verificar se existem valores Nulos e não existem (procurei em todas as colunas)


In [151]:
df['RENDA'].notnull()
#caso existisse algum valor nulo eu usaria df['RENDA'].dropna(axis=1) display(df)

0       True
1       True
2       True
3       True
4       True
        ... 
1495    True
1496    True
1497    True
1498    True
1499    True
Name: RENDA, Length: 1500, dtype: bool

4) Comando para descobrir a media desta Serie (coluna)

In [152]:
media = df['RENDA'].mean()
print(media)

984.512


5) O criterio escolhido por mim foi excluir as linhas com valores menores que 900, ja que a média é 984.512

In [153]:
coluna_renda = df['RENDA']

coluna_Nova = df[df['RENDA']>900]
coluna_Nova

ESTC  NDEP  RENDA  TIPOR  VBEM  NPARC  VPARC  TEL  IDADE  RESMS  \
2        0     0   1100      0   829      9    125    0     56     48   
3        0     0   3000      0   552     12     76    1     31     60   
4        1     0   1000      0   809     12    111    0     24      7   
8        0     0   1500      0   688      6    148    0     59     48   
10       0     0   2000      1   309      6     66    0     29      6   
...    ...   ...    ...    ...   ...    ...    ...  ...    ...    ...   
1479     3     0   1048      0   552      4    119    0     60     48   
1483     0     0   3604      1   651     24     63    0     31      6   
1485     0     0   1500      0   536      6    113    0     67     48   
1494     0     0   1100      1   412      6     67    0     41      6   
1496     1     0   2057      0   301      1    318    0     35     48   

      ENTRADA  target  
2           0       1  
3           0       1  
4           0       1  
8           0       0  
10          0       0  
...       ...     ...  
1479      119       0  
1483        0       0  
1485        0       0  
1494      100       0  
1496        0       1  

[510 rows x 12 columns]

6)

In [154]:
df.drop(['ESTC','NDEP','TIPOR','TEL','RESMS','ENTRADA','target'],axis=1,inplace=True)
df

RENDA  VBEM  NPARC  VPARC  IDADE
0       360   313      9     52     25
1       350   468     10     65     33
2      1100   829      9    125     56
3      3000   552     12     76     31
4      1000   809     12    111     24
...     ...   ...    ...    ...    ...
1495    500   437     12     60     60
1496   2057   301      1    318     35
1497    570   511     12     71     28
1498    360   384     12     52     33
1499    501   849     10    117     37

[1500 rows x 5 columns]

7)

In [191]:
import random

y = random.randint(0,1)
print(y) #valor aleatorio escolhido
for elemento in ['Renda - Idade']:
  elemento = df['Renda - Idade'] = ((df['RENDA'] - df['IDADE']) * y) 
display(df)

1


RENDA  VBEM  NPARC  VPARC  IDADE  Renda - Idade
0       360   313      9     52     25            335
1       350   468     10     65     33            317
2      1100   829      9    125     56           1044
3      3000   552     12     76     31           2969
4      1000   809     12    111     24            976
...     ...   ...    ...    ...    ...            ...
1495    500   437     12     60     60            440
1496   2057   301      1    318     35           2022
1497    570   511     12     71     28            542
1498    360   384     12     52     33            327
1499    501   849     10    117     37            464

[1500 rows x 6 columns]

8)

In [215]:

df_shuffled=df.sample(frac=1).reset_index(drop=True)
print(df_shuffled)


      RENDA  VBEM  NPARC  VPARC  IDADE  Renda - Idade
0       300   359      9     59     39            261
1       715   552      4    119     59            656
2       800   638     12     79     33            767
3       600   460      6    103     23            577
4       498   344     10     53     22            476
...     ...   ...    ...    ...    ...            ...
1495    617   464      3    116     63            554
1496   1500   849     10    117     38           1462
1497    580   460     12     62     31            549
1498   2000   984     12     83     37           1963
1499   5000   402      1    424     34           4966

[1500 rows x 6 columns]


10) como só tem colunas numéricas vou usar 2 numéricas

In [ ]:
arruma = df.groupby(['RENDA','IDADE']).count()
print(arruma)


12)

In [221]:
from google.colab import drive
drive.mount("/drive", force_remount=True)
df.to_csv('/drive/MyDrive/Trabalho 3 ENG1111/nvarqclass_analise_credito.csv')

Mounted at /drive


13)

In [222]:
from google.colab import drive
drive.mount("/drive", force_remount=True)
df.to_json('/drive/MyDrive/Trabalho 3 ENG1111/nvarqclass_analise_credito.json')

Mounted at /drive


14) https://github.com/pedroomaia/Lista3.git